pkg/provenance/provenance.go

```diff
@@ -480,48 +505,47 @@ func (o ObjectLineage) FullDiff(vNumStart, vNumEnd int) string {
 	}
 	return b.String()
 }
-func getDiff(b *strings.Builder, fieldName string, data1, data2 interface{}, vNumStart, vNumEnd int) string{
-		str1, ok1 := data1.(string)
-		str2, ok2 := data2.(string)
-		if ok1 && ok2 && str1 != str2 {
-			fmt.Fprintf(b, "Found diff on attribute %s:\n", fieldName)
-			fmt.Fprintf(b, "\tVersion %d: %s\n", vNumStart, data1)
-			fmt.Fprintf(b, "\tVersion %d: %s\n", vNumEnd, data2)
-		}
-		int1, ok1 := data1.(int)
-		int2, ok2 := data2.(int)
-		if ok1 && ok2 && int1 != int2 {
-			fmt.Fprintf(b, "Found diff on attribute %s:\n", fieldName)
-			fmt.Fprintf(b, "\tVersion %d: %s\n", vNumStart, data1)
-			fmt.Fprintf(b, "\tVersion %d: %s\n", vNumEnd, data2)
-		}
-		strArray1, ok1 := data1.([]string)
-		strArray2, ok2 := data2.([]string)
-		if ok1 && ok2 {
-			for _, str := range strArray1 {
-				found := false
-				for _, val := range strArray2 {
-					if str == val {
-						found = true
-					}
-				}
-				if !found { // if an element does not have a match in the next version
-					fmt.Fprintf(b, "Found diff on attribute %s:\n", fieldName)
-					fmt.Fprintf(b, "\tVersion %d: %s\n", vNumStart, strArray1)
-					fmt.Fprintf(b, "\tVersion %d: %s\n", vNumEnd, strArray2)
+func getDiff(b *strings.Builder, fieldName string, data1, data2 interface{}, vNumStart, vNumEnd int) string {
```

Comment:
Strange that getDiff gets included again!!

Are there any real changes that you made? If so, what I meant earlier was -- can you not include those changes in this PR?

I would like this PR to only focus on live data related changes. The formatting changes (like adding extra spaces) are fine. But if other logic changes, such as bug fixes, are included in this PR, they will get hidden under wrong commit message. And that will be problematic if later we want to go back and see where the changes are being made.

If there are no real changes to getDiff but Github is still showing it then I guess we cannot do much.

Reply:
It looks like all the code is actually the same.
I think It is just showing here because the line numbers are different due to the code at the top I added for live data?

 The good news is that func all() is here in both, and the method is the same even though it marks it as a diff

Reply:
I did not make any logic changes or bug fixes .

Reply:
I see. In that case it is fine.

Update the PR with Gopkg.toml after you test it. Then I think it will be ready for merge.

Reply:
Ok